pipeline-runner/R/seurat-3-upload_seurat_to_aws.R

```diff
@@ -15,15 +15,28 @@ upload_seurat_to_aws <- function(input, pipeline_config, prev_out) {
   scdata <- change_sample_names_to_ids(scdata, input)
   cell_sets <- get_cell_sets(scdata, input)
 
-  # add louvain clusters
-  cluster_sets <- data.frame(
-    cluster = scdata$seurat_clusters,
-    cell_ids = scdata$cells_id
-  ) %>%
-    format_cell_sets_object('louvain', scdata@misc$color_pool)
+  # detect cluster metadata
+  cluster_columns <- find_cluster_columns(scdata)
+  if (!length(cluster_columns))
+    stop(errors$ERROR_SEURAT_CLUSTERS, call. = FALSE)
+
+  # add clusters
+  cluster_sets <- list()
+  for (i in seq_along(cluster_columns)) {
+    col <- cluster_columns[i]
+
+    # first cluster column used as louvain clusters
+    method <- ifelse(i == 1, 'louvain', col)
```

Comment:
the double louvain issue I reported stems from here. There was a "louvain" column in the metadata, but it was not the first one in the metadata table.

Reply:
Yes, [this fixes it](https://github.com/hms-dbmi-cellenics/pipeline/pull/315/files#diff-6fe272316d0d7d696b8e23088d8e97aa9d24dee988ecf6ea74daac7921244d56R88)